### PR TITLE
V22F-818 adding speed limits

### PR DESCRIPTION
**Pull request by sjoeman1**
This PR adds speed limits to the track segments.
This is done in tTrackSeg. 
To test you also need the pull and build the V22F-818 branch of the client black box

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/unitTestUtilities/ComparerUtils.cpp [hunk 1]
```diff
@@ -525,6 +525,7 @@ void CompareSegments(tTrackSeg* p_segments1, tTrackSeg* p_segments2, int p_segme
             COMP_ELEM((*segOrig).type2, (*segCopy).type2)
             COMP_ELEM((*segOrig).style, (*segCopy).style)
             COMP_ELEM((*segOrig).length, (*segCopy).length)
+            COMP_ELEM((*segOrig).length, (*segCopy).length)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/unitTestUtilities/ComparerUtils.cpp:528]: Should be speedlimit

File: speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp [hunk 2]
```diff
@@ -1817,6 +1826,7 @@ void ReadTrack4(tTrack *theTrack, void *TrackHandle,
 
     curSeg = theTrack->seg;
     for(i=0; i<theTrack->nseg; i++)  {         /* read the segment data: */
+
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp:1829]: Uneccessary change to speed-dreams code

File: speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp [hunk 3]
```diff
@@ -1232,6 +1232,8 @@ CreateSegRing(void *TrackHandle, tTrack *theTrack, tTrackSeg *start, tTrackSeg *
     int		ind = 0;
 
     radius = arc = length = alf = xr = yr = newxr = newyr = xl = yl = 0;
+    // SIMULATED DRIVING ASSISTANCE: added speedLimit
+    float   speedLimit = -std::numeric_limits<float>::infinity();
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp:1236]: Pretty sure this gets overwritten by the read from the xml file, so can be removed

File: speed-dreams/source-2.2.3/data/tracks/curved-highway/curved-highway-small-4-80/curved-highway-small-4-80.xml [hunk 4]
```diff
@@ -166,6 +167,7 @@ the Free Software Foundation; either version 2 of the License, or
       <!--******************************-->
       <section name="curve 1">
         <attstr name="type" val="rgt" />
+		  <attnum name="speedLimit" val ="80" />
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/data/tracks/curved-highway/curved-highway-small-4-80/curved-highway-small-4-80.xml:170]: remove tabs

File: speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp [hunk 5]
```diff
@@ -1321,6 +1323,10 @@ CreateSegRing(void *TrackHandle, tTrack *theTrack, tTrackSeg *start, tTrackSeg *
         // TODO: is the (int) intended?
         DoVfactor = (float) ((int) GfParmGetCurNum(TrackHandle, path, TRK_ATT_DOVFACTOR, (char*)NULL, 1.0)) ;
 
+        //SIMULATED DRIVING ASSISTANCE: added speed limit
+        // if no speed limit is defined it will be infinity
+        speedLimit = GfParmGetCurNum(TrackHandle, path, TRK_ATT_SPEEDLIMIT, (char*)NULL, std::numeric_limits<float>::infinity());
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp:1328]:
```suggestion
        speedLimit = GfParmGetCurNum(TrackHandle, path, TRK_ATT_SPEEDLIMIT, nullptr, std::numeric_limits<float>::infinity());
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/data/tracks/curved-highway/curved-highway-small-4-80/curved-highway-small-4-80.xml (hunk 4 above)
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/data/tracks/curved-highway/curved-highway-small-4-80/curved-highway-small-4-80.xml:170]: forgot to change tabs in Cas's IDE, will be changed soon

**Review by suppergerrie2 — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp [hunk 6]
```diff
@@ -1321,6 +1323,10 @@ CreateSegRing(void *TrackHandle, tTrack *theTrack, tTrackSeg *start, tTrackSeg *
         // TODO: is the (int) intended?
         DoVfactor = (float) ((int) GfParmGetCurNum(TrackHandle, path, TRK_ATT_DOVFACTOR, (char*)NULL, 1.0)) ;
 
+        //SIMULATED DRIVING ASSISTANCE: added speed limit
+        // if no speed limit is defined it will be infinity
+        speedLimit = GfParmGetCurNum(TrackHandle, path, TRK_ATT_SPEEDLIMIT, nullptr, std::numeric_limits<float>::infinity());
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/track/trackv1/track4.cpp:1328]:
```suggestion
        float speedLimit = GfParmGetCurNum(TrackHandle, path, TRK_ATT_SPEEDLIMIT, nullptr, std::numeric_limits<float>::infinity());
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]

**Review by larsdekwant — CHANGES_REQUESTED** [review]
See my old change requests

**Review by larsdekwant — APPROVED** [review]
Works correctly when used with the updated, but requires an updated BlackBox. Otherwise the interventions will be random. Needs to be merged together with the PRs on the other repositories.

**Review by KungFuDonkey — APPROVED** [review]

**Pull request merged**